sonar-db/src/main/java/org/sonar/db/version/v60/PopulateProfileKeyOfActivities.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * SonarQube
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.db.version.v60;
+
+import java.sql.SQLException;
+import java.util.Map;
+import org.sonar.api.utils.KeyValueFormat;
+import org.sonar.db.Database;
+import org.sonar.db.version.BaseDataChange;
+import org.sonar.db.version.MassUpdate;
+import org.sonar.db.version.Select;
+import org.sonar.db.version.SqlStatement;
+
+public class PopulateProfileKeyOfActivities extends BaseDataChange {
+
+  public PopulateProfileKeyOfActivities(Database db) {
+    super(db);
+  }
+
+  @Override
+  public void execute(Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("select id, data_field from activities where profile_key is null");
+    massUpdate.update("update activities set profile_key=? where id=?");
+    massUpdate.rowPluralName("activities");
+    massUpdate.execute(PopulateProfileKeyOfActivities::handle);
+  }
+
+  private static boolean handle(Select.Row row, SqlStatement update) throws SQLException {
+    int id = row.getInt(1);
+    String data = row.getString(2);
+    Map<String, String> fields = KeyValueFormat.parse(data);
+
+    update.setString(1, fields.get("profileKey"));
```

Comment:
Is it possible that there's no profileKey ?
I think we should fail fast here, otherwise the next migration that set the profile column to not nullable will fail but it will harder to understand from where.
Failing here will allow to know the id of the row.